### ISSUE TITLE
Realtime for volcanic ash

### ISSUE THREAD

**Issue by Charlotte-Morgan** [inasafe/inasafe#2491]
# Background

Geoscience Australia has been working technical agencies in Indonesia to develop a volcanic ash modelling capability. After being shown InaSAFE, the volcanic ash modelers are keen to find ways to share information with DM and communities and to share hazard data for situational awareness during a volcanic eruption.

InaSAFE version 3.4 will include a new IF for ash on land cover.
there is also a request for a real-time interface.
# proposal
- [ ] volcanic ash hazard data to be provided by technical agency
- [ ] new IF for ash impact on crops - calculate area affected
- [ ] ash impact on people - number of people affected (no fatality calculation for this)
- [ ] temporal analysis, time slices with impact of cumulative ash build up over time - NEW
# impact map

First draft of what an impact map might look like - note this is a static pdf :(
![image](https://cloud.githubusercontent.com/assets/5743135/11528720/0eb7314a-991b-11e5-98d9-ccb843fc83e0.png)
# comments

There could be many similarities between flood and volcanic ash (not really), ie in the way the data may be supplied and the type of analysis and presentation of impact analyses. eg not a single event like an earthquake; more likely to be identified by a time stamp; time series analysis or reporting will be required; compounding effect of ash buildup or prolonged duration of inundation
# user requirements

Workshops and meetings will be held to determine more specific user requirements. Please add any details that are needed:
- [ ] which organisation will provide hazard data and in what format
- [ ] which organisation will host the real time system
# cc

@lucernae @samnawi @timlinux

**Comment by timlinux** [comment]
@Charlotte-Morgan thanks for making a report mockup - that is very useful. We will stand by for further details on requirements from your volcano colleagues.

**Comment by fredychandra** [comment]
HI @Charlotte-Morgan 
I think It will be nice if we can also adding how many crops and settlements area will be covered by ash potentially in that report.

**Comment by samnawi** [comment]
:+1: for @fredychandra suggest to add the impacted for crops and settlements area. The other important thing is also the impact for the airport, as we know when the volcanic eruption happens,one of the issue is about air transportation.

**Comment by Charlotte-Morgan** [comment]
This proposal is being driven very much by the technical and user community and I have simply documented what they have proposed.
They will be holding workshops to clarify user requirements; the first is scheduled for 27 January - 5 February in Melbourne. It will involve GA, BOM, and volcano experts from BG.
From my discussions with them; impact on crops is something they are interested in & impact on aviation is something they are keen to avoid.
Please keep your ideas and suggestions coming as we continue to clarify user requirements.

**Comment by adelebearcrozier** [comment]
Hello all,
I am the volcanologist hosting the workshop mentioned by Charlotte next week for two weeks (25/1-5/2) with invited volcanologists from Badan Geologi. We will focus on familiarising ourselves with the work done so far in the volcano space with InaSAFE, scoping the volcanic ash hazard data requirements and refining the report mock-up for the "volcanic ash impact map" that we have already provided with a view to initiating dispersion model runs which can be used for further product development. We will also agree upon timing for a subsequent sprint of work in Indonesia (likely May) where we will finalise the hazard data. I will update you on the outcomes in a couple of weeks time. 
Kind regards,
Dr Adele Bear-Crozier

**Comment by Charlotte-Morgan** [comment]
@heriwaseo
welcome to the ticket we have for Realtime ash discussion. I still need to add the information @adelebearcrozier provided from your workshop :)

**Comment by Charlotte-Morgan** [comment]
I am sharing the information received from @adelebearcrozier following the ash modelling workshop held in Melbourne in January 2016. Anjar, Nugi and Adele put some considerable thought on the type of volcanic ash impact analysis and how they would like to share this with the community. Please note: we are talking about volcanic ash impact on the ground and not on flight paths or aviation.

The following is from the official trip report
" As part of the DMInnovation work plan for FY 15/16 the team work shopped a concept and design for a “Volcanic Ash Impact” map to be generated through the InaSAFE real time platform. The concept for this product is based on the existing Shake Map product generated by BMKG for earthquake impact and made publicly available through the InaSAFE real-time earthquake platform. The design activity focused on format of the map product, content (which included delineation of fallout levels and their potential impact on communities) and simple and straightforward communication of scientific modeling outputs to the general public. The team put particular emphasis on the importance of providing estimates on area of land cover affected (e.g. rice fields, plantations, settlement) in order to engage communicates impacted by volcanic ash. Two graphical mock-ups of the proposed product were finalized, one in English and one in Bahasa Indonesian (see Figure below). Documentation to accompany the two mock-ups was also drafted further detailing the components of the product. Example hazard data in the form of forecast scenarios for high profile Indonesian volcanoes was generated for testing purposes throughout development of the product. The mock-ups, documentation and hazard data are to be provided to the InaSAFE development team through DMInnovation for consideration/development.  
The team also considered the web-interface for the “Ash Map” which will presumably sit alongside the “Shake Map” and “Flood Map” on the real-time platform. Documentations regarding recommendations for the “Ash Map” page will be forwarded to DMInnovation. " .. thank you for this report.

# Impact Maps etc
[VAIM_EN copy.pdf](https://github.com/inasafe/inasafe/files/137479/VAIM_EN.copy.pdf)
[VAIM_BI copy.pdf](https://github.com/inasafe/inasafe/files/137480/VAIM_BI.copy.pdf)
 # Annotation
[VAIM_annotation_1 copy.pdf](https://github.com/inasafe/inasafe/files/137481/VAIM_annotation_1.copy.pdf)
[VAIM_annotation_2 copy.pdf](https://github.com/inasafe/inasafe/files/137482/VAIM_annotation_2.copy.pdf)
[VAIM_annotation_3 copy.pdf](https://github.com/inasafe/inasafe/files/137483/VAIM_annotation_3.copy.pdf)

# Sample data
The team has also provided sample data for volcanic ash.
It is available as a continuous raster ascii file with approx 1km cell size.
@samnawi is working on converting to a classified raster in tiff format

# proposed action
@samnawi - please work on the hazard data conversion and ask the VA team if you have any questions, especially about class breaks.
@samnawi  - please review the BIG landcover data and check with the VA team if you have any questions about attribute mapping
@samnawi - prepare an airport infrastructure data set for Indonesia
@samnawi - finish the new population data for districts :) as per eq realtime
@fredychandra - any thoughts on how to report landcover impact?
@adelebearcrozier - please share any other feedback or ideas you have

# cc
@timlinux @lucernae

**Comment by samnawi** [comment]
## 1. Data Conversion and classification
> @samnawi - please work on the hazard data conversion and ask the VA team if you have any questions, especially about class breaks.

- Convert from .asc to .tif
- Because the availability of Impact function in InaSAFE just for _Classified Hazard_, so the hazard must be classified in the keyword. There is a problem if we not reclassified it first like in this picture.
![screenshot_21](https://cloud.githubusercontent.com/assets/8871958/13241552/b1a23384-da1e-11e5-9b46-5dec583ff04f.png)
InaSAFE will read all of the unique value in the hazard --> hard (need extra effort) to classified (drag the value one by one)

- As the classified that has been shared by @adelebearcrozier, hazard data was reclassified into 5 class.
![screenshot_22](https://cloud.githubusercontent.com/assets/8871958/13241889/e4c59a32-da21-11e5-843f-07b24fe5a478.png)
And then classified it.
![image](https://cloud.githubusercontent.com/assets/8871958/13241928/4328ec28-da22-11e5-9285-f1b45e5dc7f0.png)
@adelebearcrozier @heriwaseso Could we classify the hazard like in the picture in above? Because InaSAFE just provides 3 class for the hazard, or maybe you have any suggestion to classified it?
@timlinux @lucernae Is it possible to make InaSAFE have 5 class also (if the classification from Badan Geologi becomes an agreement or reference)

**InaSAFE Testing**
![image](https://cloud.githubusercontent.com/assets/8871958/13242452/4bd4d2e2-da27-11e5-9c82-e0f539941327.png)

**The Result**
![image](https://cloud.githubusercontent.com/assets/8871958/13242419/06f78052-da27-11e5-953c-9724c3fa9b3f.png)

**Proposed in this part**
Could we make a continuous hazard in this impact function? Because I see the same parameter in the option of volcanic ash with the option in earthquake or tsunami
![image](https://cloud.githubusercontent.com/assets/8871958/13242206/163fe4c0-da25-11e5-9812-7968870e19b0.png)

## 2. Prepare some data
> @samnawi - prepare an airport infrastructure data set for Indonesia
@samnawi - finish the new population data for districts :) as per eq realtime

I've been downloaded airport data from http://indonesia.tanahair.go.id for the whole of Indonesia.
For the population data, I've shared in this link (https://www.dropbox.com/sh/ibhodjntetcfcg1/AABAtobVTWZ3sxeTOD0rPVLpa?dl=0)

Thank you.

**Comment by adelebearcrozier** [comment]
Re: 5 classes versus 3 classes. If possible we would prefer to keep the 5 classes as the breaks in these classes are following conventions being set internationally and we want to be consistent. Happy to discuss further.

**Comment by Charlotte-Morgan** [comment]
Hi Martin -  @wonder-sk - welcome to the discussion. This ticket has volcanic ash realtime user requirements. Martin will be working up the volcanic ash on land cover IF. This will be the first implementation of landcover analysis.

**Comment by Charlotte-Morgan** [comment]
Hi @adelebearcrozier 
I have asked the team here to review the ash data your team developed. I have a couple of questions now that we are getting the developers on board with this one.
The ash data you  supplied is raster with 1km cell size. Do you anticipate the final ash data to be the same or would it be possible to provide something with a smaller cellsize or maybe vector data ie contoured data with breaks as per the classes specified above?
From my perspective, we can change InaSAFE to work with 5 classes, should be no problem.
I note that our initial landcover impact function has been designed to use vector hazard data #1216

**Comment by timlinux** [comment]
Hi @adelebearcrozier 

We are doing our work planning for ASH realtime. I think the first port of call is for us to understand the hazard data. Could you provide some details about where ash data is published? Also will additional details such as plume height be available with the ash data feed?

**Comment by timlinux** [comment]
For ease of reference I am adding the above PDF reports as images here:

<img width="574" alt="screen shot 2016-04-06 at 10 17 37" src="https://cloud.githubusercontent.com/assets/178003/14305260/b814ba88-fbe4-11e5-8d1d-46ae74802caa.png">

![Uploading Screen Shot 2016-04-06 at 10.31.52.png…]()
<img width="1237" alt="screen shot 2016-04-06 at 10 30 42" src="https://cloud.githubusercontent.com/assets/178003/14305276/ddf29e00-fbe4-11e5-9024-ff0912260793.png">
<img width="1238" alt="screen shot 2016-04-06 at 10 30 23" src="https://cloud.githubusercontent.com/assets/178003/14305277/e0fd1d78-fbe4-11e5-80ae-b92335877840.png">

**Comment by timlinux** [comment]
We had a good meeting with @Charlotte-Morgan @fredychandra and @samnawi today, so here is my take on the requirements in terms of data inputs:

Dataset | Notes | Source
------------ | ------------- | --------
Event ID | We need an event id to uniquely reference each report. It could simply be YYYY-MM-DD-HH. | Generated if not available externally
Alert level | This is the general volcano hazard alert level | Possibly scrape from the [Badan Geologi web site](http://vsi.esdm.go.id/index.php/gunungapi/aktivitas-gunungapi)
Column height | This is the current height of the ash column above the volcano at time of reporting | **Uncertain where this will come from**
Location description | This includes x,y position, name, relative position, province | We can do this once-off and store in our volcano table
Date | This is the report date | Is should be supplied with the ash model data (perhaps encoded in the url or dataset name)
Elapsed time | Time since data was created that report was generated | Computed programmatically
Landcover | This is 1:25k land cover data with 5 lands classes as described above | IGD 25k cleaned by @samnawi 
Place names | These will be region capitals correctly geolocated with population counts for urban dwellers. Same dataset to be used for earthquake realtime. | @samnawi has prepared this based on census data.
Volcanos | Standard **point** dataset for all volcanos.Note the additional description attributes needed. | @samnawi will supply
Hazard zones | Will be extracted (or more ideally supplied as polygons) from the ash model. The data will be declassed into thickness classes as per table 1. | **Currently it is not clear how to obtain this data.**
DEM | Used for backdrop - we will generate a hillside from it in greyscale. @samnawi to take care of this. | SRTM via @samnawi . Apparently ASTER data is also free now so may want to consider using that.
Airports | Need airport name and point location | @samnawi

**Comment by adelebearcrozier** [comment]
Hi @timlinux , @Charlotte-Morgan , @fredychandra and @samnawi ,
Its great to see all of this progress. I will attempt now to answer your questions but i am available on skype to flesh them out further:

@Charlotte-Morgan - " _The ash data you supplied is raster with 1km cell size. Do you anticipate the final ash data to be the same or would it be possible to provide something with a smaller cellsize or maybe vector data ie contoured data with breaks as per the classes specified above_?"

We generate ash hazard data initially in netCDF (all variables over all time periods). Typically we dont do anything further as we have python plotting tools for analysing and visualising the data direct from netCDF. However, when interfacing with GIS in the past we typically extract as raster data. 1km is the default cell size we use however we output finer and coarser cell sizes depending on the scale of the eruption and anticipated degree of impact (e.g. short range, building impact - finer res; long range aviation impact - coarser res. We don't generate vector data.  

@timlinux - _Could you provide some details about where ash data is published? Also will additional details such as plume height be available with the ash data feed?_

Currently the ash hazard data is generated either in the lead up to or during and eruption by Badan Geologi volcanologists with technical support from Geoscience Australia. The model is hosted on a number of high performance computing infrastructures (e.g. raijin (NCI), smong (DM Innovations), Volcanic Ash Advisory Center (VAAC)) and the data currently is archived on Smong (Jakarta). We can make it available in a range of file formats. We can also ensure that additional details such as "column height" and "alert level" are included in the naming convention for the ash data files. I have attached a volcano table we use and maintain for this region which includes volcano name, lat/lon, summit elevation, region, sub-region etc and is current to 2016 if you think this would be helpful also. 

ABC
[GVP_Volcano_List_Darwin_VAAC_AOR_final.xlsx](https://github.com/inasafe/inasafe/files/206234/GVP_Volcano_List_Darwin_VAAC_AOR_final.xlsx)

**Comment by samnawi** [comment]
I've bee given @lucernae the landcover data at 1 : 250,000 (national level) and also the location of the airport. All of the data was downloaded from http://tanahair.indonesia.go.id/home/

Thank you

**Comment by Charlotte-Morgan** [comment]
Thanks Adele.
Iwan will continue to discuss ash data formats etc with you and will liaise with developers.

Charlotte  Morgan
Spatial Data Analyst
DMInnovation - Jakarta
GEOSCIENCE AUSTRALIA

Mobile: +61 412 949642
Mobile: +62 8118204648


On 6 Apr 2016, at 5:48 PM, Dr Adele Bear-Crozier <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi @timlinux<https://github.com/timlinux> , @Charlotte-Morgan<https://github.com/Charlotte-Morgan> , @fredychandra<https://github.com/fredychandra> and @samnawi<https://github.com/samnawi> ,
Its great to see all of this progress. I will attempt now to answer your questions but i am available on skype to flesh them out further:

@Charlotte-Morgan<https://github.com/Charlotte-Morgan> - " The ash data you supplied is raster with 1km cell size. Do you anticipate the final ash data to be the same or would it be possible to provide something with a smaller cellsize or maybe vector data ie contoured data with breaks as per the classes specified above?"

We generate ash hazard data initially in netCDF (all variables over all time periods). Typically we dont do anything further as we have python plotting tools for analysing and visualising the data direct from netCDF. However, when interfacing with GIS in the past we typically extract as raster data. 1km is the default cell size we use however we output finer and coarser cell sizes depending on the scale of the eruption and anticipated degree of impact (e.g. short range, building impact - finer res; long range aviation impact - coarser res. We don't generate vector data.

@timlinux<https://github.com/timlinux> - Could you provide some details about where ash data is published? Also will additional details such as plume height be available with the ash data feed?

Currently the ash hazard data is generated either in the lead up to or during and eruption by Badan Geologi volcanologists with technical support from Geoscience Australia. The model is hosted on a number of high performance computing infrastructures (e.g. raijin (NCI), smong (DM Innovations), Volcanic Ash Advisory Center (VAAC)) and the data currently is archived on Smong (Jakarta). We can make it available in a range of file formats. We can also ensure that additional details such as "column height" and "alert level" are included in the naming convention for the ash data files. I have attached a volcano table we use and maintain for this region which includes volcano name, lat/lon, summit elevation, region, sub-region etc and is current to 2016 if you think this would be helpful also.

ABC
GVP_Volcano_List_Darwin_VAAC_AOR_final.xlsx<https://github.com/inasafe/inasafe/files/206234/GVP_Volcano_List_Darwin_VAAC_AOR_final.xlsx>

-

**Comment by timlinux** [comment]
Hi @adelebearcrozier 

Just a note to say thanks for the volcanos table you provided - that will indeed be useful.

In terms of moving forward, I think it might be good to schedule a time to chat on Skype if you are available. I am on GMT+2 (South Africa time) for the next few weeks and @lucernae & @samnawi  are on GMT+7 - so it might be difficult to find a time where one of us doesn't have to stay up a while. Charlotte is currently on leave but I would like to keep things moving along in her absence if possible. I have made a doodle here to try to find a convenient time:

http://doodle.com/poll/hgr5g4szequn5tk8

@samnawi Could we make a shared folder (e.g. google drive) where we assemble the various datasets outlined above? It will make it easier if we have everything collated in one place.

@adelebearcrozier Could you provide us with some sample data (original netcdf is fine)?

> @adelebearcrozier wrote: 1km is the default cell size we use however we output finer and coarser cell sizes depending on the scale of the eruption and anticipated degree of impact (e.g. short range, building impact - finer res; long range aviation impact - coarser res. We don't generate vector data.


* @adelebearcrozier : Are these all generated from the same raw NetCDF products, or is the NetCDF itself generated at the scale suitable to the use purpose as outlined above?

* @adelebearcrozier : Also are you able to share the python code that you use for generating products, and is it available under an open source license? Is there any value in us having access to the code (e.g. to extract our own derivative products? 

* @adelebearcrozier In the mockups provided above the hazard data is illustrated as nice contoured data. You mention above not generating any vector products. Do you have any philosophical  / technical issue with us using the raster to do some simple transformations in the form of raster -> recalls to hazard classes -> extract polygonal areas for each class -> perform a polygon on polygon (ash hazard zones on landcover)? Or did you have any or preconceived ideas of how to best use the data in an analysis? For classification / zonation of the ash data we would use the ash thickness ranges as per your diagram above.

* @adelebearcrozier in terms of dissemination of the hazard data, you mentioned a bunch of institutions that curate the data. Are these all maintaining copies of the same datasets, or are different institutions each holding a subset of the total available data? Where is the best place for us to download the data? Smong? If they have a substantial archive it would be very useful - we could back generate historical reports and that would give us the ability to test the system we build under a wide range of input data. If the data is not publicly disseminated, perhaps we can look at an alternative approach whereby we provide an upload page for NetCDF products. When an upload takes place we automatically process it and generate a report. This implies somewhat manual intervention in the process - I guess it would be better if things can be fully automated, but my impression from your comments above is that this (full automation) may not match the actual real world workflows?

With apologies for a rather long comment in this issue :-)

**Comment by samnawi** [comment]
Hi @timlinux I've prepared some data in google drive. You can access the folder in [here](https://drive.google.com/open?id=1WXNpgn556dsP7sUBKFfReDflW6nHd-eD198OhJBfTJw).

Note :
If I'm not wrong, for land cover data, we use at 1 : 250,000 for the national level not 1 : 25,000. It's also according to 1km x 1km for from ash hazard pixel size.

I still prepare the DEM / hill shade data for the background. Because the RAW data for DEM is too big size to shared.

Also I'll update the of the airport, because in data that I shared, has incomplete name for the airport.

Thank you.

**Comment by timlinux** [comment]
Hi @samnawi - thanks for this. For convenience here is the catalogue data:

<img width="577" alt="screen shot 2016-05-01 at 22 42 47" src="https://cloud.githubusercontent.com/assets/178003/14944265/309965ae-0fee-11e6-806d-743be62a87a8.png">


Can you share the actual folder link? The above link only takes me to the catalogue spreadsheet.

Thanks!

**Comment by adelebearcrozier** [comment]
Hi @timlinux 

My responses to your questions above: Can discuss further when we Skype.

_@adelebearcrozier Could you provide us with some sample data (original netcdf is fine)?_

Yes – please find a sample netCDF file for Soputan volcano on my GoogleDrive here: https://drive.google.com/open?id=0B9XeBru2QyndUmF4NjBvQUJrVXc (let me know if you have any issues accessing). This file contain all the variables generated by PF3D for a standard forecast run. The Ash Impact Map is only concerned with the variable “Ground deposit thickness”. 
@adelebearcrozier wrote: 1km is the default cell size we use however we output finer and coarser cell sizes depending on the scale of the eruption and anticipated degree of impact (e.g. short range, building impact - finer res; long range aviation impact - coarser res. We don't generate vector data.

_@adelebearcrozier : Are these all generated from the same raw NetCDF products, or is the NetCDF itself generated at the scale suitable to the use purpose as outlined above?_

The netCDF is generated in the required spatial resolution. If we wanted to change the resolution we would need to generate a new netCDF. 

_@adelebearcrozier : Also are you able to share the python code that you use for generating products, and is it available under an open source license? Is there any value in us having access to the code (e.g. to extract our own derivative products? _

The code for extracting the raster data from the raw netCDF output is available as part of the 	open source PF3D distribution which we host on github here: 	
https://github.com/GeoscienceAustralia/PF3D – you are welcome to it and I can help you find the necessary script if needed. Originally we used it get raster data out in various formats to visualise in GIS. We use this functionality less and less these days as we can generate plots quickly and efficiently directly from the netCDf using python plotting tools like matplotlib etc. I am currently developing an 	ipython notebook for BG volcanologists which takes in the raw netCDF and outputs a series of 	useful map products and animations for analysis. I will include this in the next version of PF3D. 

_@adelebearcrozier In the mockups provided above the hazard data is illustrated as nice contoured data. You mention above not generating any vector products. Do you have any philosophical / technical issue with us using the raster to do some simple transformations in the form of raster -> recalls to hazard classes -> extract polygonal areas for each class -> perform a polygon on polygon (ash hazard zones on landcover)? Or did you have any or preconceived ideas of how to best use the data in an analysis? For classification / zonation of the ash data we would use the ash thickness ranges as per your diagram above._

No problems with this. It sounds great to me.

_@adelebearcrozier in terms of dissemination of the hazard data, you mentioned a bunch of institutions that curate the data. Are these all maintaining copies of the same datasets, or are different institutions each holding a subset of the total available data? Where is the best place for us to download the data? Smong? If they have a substantial archive it would be very useful - we could back generate historical reports and that would give us the ability to test the system we build under a wide range of input data. If the data is not publicly disseminated, perhaps we can look at an alternative approach whereby we provide an upload page for NetCDF products. When an upload takes place we automatically process it and generate a report. This implies somewhat manual intervention in the process - I guess it would be better if things can be fully automated, but my impression from your comments above is that this (full automation) may not match the actual real world workflows?_
	
The netCDF file (data) will be stored on Smong but this is not publically available. I think your option above to have upload page would be perfect. I think initially BG would want 	manual over site of report generation so that they can fully understand what is going out to the 	public and move gradually to a more automated approach.  

Cheers
Adele

**Comment by timlinux** [comment]
Hi @adelebearcrozier 

OK that is great feedback thanks. @lucernae So it would be a good first task for you to write some utilities / helper functions to:

* take a netCDF file as input
* extract the needed sublayer
* reclassify the layer into five hazard classes as per the legend above
* run gdal_polygonize on the layer to produce a vector layer where each polygon is assigned a class
* add 3 columns to the vector layer which have:
  * the human readable name of the hazard level
  * the description of the hazard level (maybe we don't need this)
  * the numeric range of the hazard level
* set the keywords of the vector layer (purpose: hazard, hazard: ash)

Use the same approach as the grid.xml convertor - i.e. a little dialog fronted that lets you select an input file and an output file name, and then some re-usable backend code that does the actual conversion.

Once we have that in place we can create a new impact function modelled on flood polygons on landuse.

As a little R&D side project it would be worth looking to see if we can smooth the polygon edges nicely - since we need similar routines for doing that with earthquake (see #2662). We would just need to make sure the polygon topology is maintained after smoothing. What do others think about doing this?

**Comment by timlinux** [comment]
@lucernae it might make sense for you and @Gustry to work together on the polygoniser stuff

**Comment by Charlotte-Morgan** [comment]
Hi Everyone
I'm adding @ivanbusthomi to this conversation. please welcome Ivan, he recently joined the team at DMInnovation as a GIS  Officer and once he is up to speed, will take an active role in all the realtime projects

**Comment by samnawi** [comment]
I'm sorry.

I just realized if I shared the wrong link.
This is the right link --> https://drive.google.com/folderview?id=0B3b8OAVI_hAiVjFaSVhhZTVwTDQ&usp=sharing

Thank you.

**Comment by ivanbusthomi** [comment]
Hi everyone, i'm looking forward to work with you all. I will do my best :smile:

**Comment by timlinux** [comment]
Just an update here for everyone watching this thread: I did some processing to generate vector products from the supplied netcdf example above. This is how the products look:

<img width="691" alt="screen shot 2016-05-24 at 17 54 56" src="https://cloud.githubusercontent.com/assets/178003/15511518/30adec36-21dc-11e6-8e22-870d1aeb0aaa.png">


This is the raster product that I produced using the following methodology:



Extract the thickness band from the sample data set:

```
/Library/Frameworks/GDAL.framework/Versions/1.11/Programs/gdal_translate \
  -b 65 \
  -of Tiff \
  NETCDF:"/Users/timlinux/Downloads/soputan_forecast_2016.res.nc":THICKNESS thickness.tif
```

Use the QGIS raster calculator to derive a classified product according to these classes (as provided in your screenshot above):

![Uploading Screen Shot 2016-05-24 at 17.40.55.png…]()


This is the expression I used:

```
(0 * "Thickness@1" = 0) + 
(1 * "Thickness@1" > 0.01 and  "Thickness@1" < 0.1) + 
(2 * ("Thickness@1" >= 0.1 and "Thickness@1" < 2)) +  
(3 * ("Thickness@1" >= 2 and "Thickness@1" < 5)) +  
(4 * ("Thickness@1" >= 5 and "Thickness@1" < 10)) +  
(5 * ("Thickness@1" >= 10))
```

This produced an output raster with 6 classes (including 0) which I attach below for your interest:

[Uploading thickness.tif…]()

I used GDAL's polygonize tool to extract a vector dataset with one polygon per zone:

```
gdal_polygonize.py /private/tmp/thickness.tif -f "ESRI Shapefile" /tmp/ash.shp ash level
```

This is the resulting dataset:

[Uploading volcano.zip…]()

**Comment by timlinux** [comment]
One other comment I should add is that the NetCDF has multiple time slices for each product - can we assume that we use the 90h product for analysis?

**Comment by adelebearcrozier** [comment]
Hi @timlinux, Many thanks. This looks excellent. I am currently at DM Innovation (leaving tonight) and have had good discussions with the team (@Charlotte-Morgan @samnawi @ivanbusthomi). I am now working on producing netCDF test files for multiple scenarios which i will store in /model_area/tephra/inasafe_realtime_hazard-data for @samnawi. 

In response to your question, yes the 09th time step is the correct one but i am conscious that 90 time steps is definitely not practical and i am attempting to create just a single time step (the final one) to cut down on the file size and complexity. This file of one time step will likely include all the variables still (including thickness) as removing the other variables will be a more complicated undertaking and i would rather use a script to extract what InaSAFE needs and use the same netCDF for other purposes rather than generate lots of netCDF's in different configurations (something i know Badan Geologi wont do). 

We have also had further discussion regarding the resolution and we seem to be circling back to using 1km. Although this is better for the land cover analysis there are challenges with this such as no georeferencing (we run on a utm grid for the 1km as opposed to lat lon for the 12km grids which are easier to georeference) although @Charlotte-Morgan thinks we just need to project the data in WGS84 and all will be good. We can discuss further when we next skype. 

I am also working on providing answers to questions regarding the break points in the hazard data classification, the grouping of land cover types into the land cover schema being used for the map and our requirements for the web interface.

**Comment by Charlotte-Morgan** [comment]
thanks @adelebearcrozier for the update. 
It was great to meet face to face and chat about a few things. 
DMI also showed off realtime UI and landing page and interoperability with InAWARE
Here are a few notes from me and actions.
**Data**: Adele is preparing more sample data so that @samnawi will run landcover IF and see what different scale events look like; and provide feedback.
**Scale**: default projections are WGS84 for 12km netcdf files and UTM for 1km netcdf files. Note a 1km file will be limited to a single UTM zone

**Adele to provide more info on:**
- Break points: need to be provided to more decimal places
- Lancover classes: DMI need full mapping between BIG data and BG classes or a reference or both
- Realtime landing page: text, seach filters, downloads etc

**Action list** for volcanic ash on landcover should include questions about:
- Do you need to cover any crops or water before the ash fall?
- Which airports will be affected / open for response / evacuations?
**Notes and assumptions:** should list break points in the analysis - these should be dynamic in case user has changed the options @easmetz - this is relevant to your review of all reports

@adelebearcrozier (is there a depth of ash at which an airport will be closed at ground level?)

**Key words Wizard:** volcanic ash data are in cm - we need to update the units and parameters in the keywords wizard to cope with cm and or meters to 4 decimal places) ie 0.02 cm or 0.0002 m.

**Default break points**: we need to set default break points for volcano ash hazard classes 

**Other actions from meeting:**
- DMI will prepare a demo InaSAFE kit for Adele
- DMI will socialise new InaSAFE / Ash functionality with BG team

**Comment by adelebearcrozier** [comment]
Hi @samnawi, i have created 1km netCDF files for five volcanoes (lokon, merapi, soputan, sinabung and sangeang_api) and placed then on the drive here:

/model_area/tephra/inasafe_realtime_hazard-data/netcdf

Please let me know if you have any trouble accessing them. Please only consider the variable "THICKNESS" and the final time step. I will look into generating a single time step next week. 

Great to have met with you all today, many thanks. Cheers.

**Comment by ivanbusthomi** [comment]
Hi everyone,
Yesterday I had a fun playing with netcdf conversion using Tim's method.
> /Library/Frameworks/GDAL.framework/Versions/1.11/Programs/gdal_translate \
>   -b 65 \
>   -of Tiff \
>   NETCDF:"/Users/timlinux/Downloads/soputan_forecast_2016.res.nc":THICKNESS thickness.tif

This is the repository of what i did yesterday. https://github.com/ivanbusthomi/cdf_converter . It is a qgis plugin to convert the netcdf file into a geotiff file. I did it to help me convert the netcdf file so i can use it as an input in inasafe desktop analysis.
![main_dialog](https://cloud.githubusercontent.com/assets/7351977/16142349/aa918d04-348e-11e6-92eb-71050bdfbaa6.PNG)

I am not aware whether there is already a function to do the conversion in inasafe or not, so I apologize if what i did is a duplicate of someone else's work :sob: .
Feel free to give some feedback.
___
**CC** @Charlotte-Morgan @adelebearcrozier @timlinux @fredychandra @samnawi

**Comment by adelebearcrozier** [comment]
thanks for the update @ivanbusthomi 
Can i ask what 'select band' means?
I'm assuming its timestep??
Cheers

**Comment by adelebearcrozier** [comment]
Hi @Charlotte-Morgan  - would you mind adding Rikki to this ticket?? As the new Activity Leader i would like her to be able to see the progress on this work. Cheers.

**Comment by Charlotte-Morgan** [comment]
Hi @adelebearcrozier - sure. Lets add @RikkiWeber @easmetz and @lrpower .
Rikki is the GA Indonesia program activity leader who should be aware if not involved in our work, Libby who is working on some components of InaSAFE project; its always good to see a well written, active ticket; and Lauren who will visit from GA to look after the spatial team in Jakarta while I am on leave.

**Comment by ivanbusthomi** [comment]
Hi @adelebearcrozier sorry for the late reply
Yes, the band is actually the time step. There are few improvement in the plugin that I'm about to update soon.

**Comment by timlinux** [comment]
@lucernae can you give a status update on the ASH IF implementations please?

**Comment by Charlotte-Morgan** [comment]
hi @adelebearcrozier - quick update for you on this work. Indonesia has been on holiday for the last week. @lucernae will join his Kartoza colleagues at DMI tomorrow. We have a 2 day ASH hack planned for Wednesday and Thursday this week.  More specifically our schedule for the site visit says: ASHamopalooza 3.5 "We have to make ASH IF's rock! And ASH realtime!"
If you have any specific questions or concerns, please let us know.
@RikkiWeber - FYI

**Comment by Charlotte-Morgan** [comment]
Hi @ivanbusthomi - can you please put the extracted data from the netcdf files in the minion data please.
Ie - extract the relevant data from the sample data at
 \\DS1813-NAS\model_area\tephra\inasafe_realtime_hazard-data\netcdf
and put it here
C:\AIFDR\Minion_Data\09_Projects\160215 Realtime Ash Data

**Comment by Charlotte-Morgan** [comment]
@adelebearcrozier - just reviewing the actions from our discussion on May 25th. 
Could you please help us with any advice on the following points we discussed: 
- Break points: need to be provided to more decimal places
- Landcover classes: DMI need full mapping between BIG data and BG classes or a reference or both
- Realtime  landing page: text, seach filters, downloads etc
thanks heaps

**Comment by ivanbusthomi** [comment]
Hi @Charlotte-Morgan 
I already put the extracted NetCDF somewhere in the Minion Data. I don't really remember which directory I put it in since I can't access the Minion Data from my laptop. I will update it once I got back to the office.

**Comment by ivanbusthomi** [comment]
Hi @adelebearcrozier, there are a few questions from me regarding the ash report template
___
### Sorting order
I want to confirm the sorting order of affected city and airport in the report template. Here is a screenshot of Merapi's data
![places_sort_order](https://cloud.githubusercontent.com/assets/7351977/16862734/8af80fe2-4a76-11e6-9ad1-a88a70b93c78.PNG)

As we can see, there are several places and airports that are affected in different fallout levels. According to Table 2 in the reference below, **the prefered order** that will be shown in the report table are descending from high fallout level to the low one, right?
![Table2](https://cloud.githubusercontent.com/assets/178003/14305276/ddf29e00-fbe4-11e5-9024-ff0912260793.png)

So, using that reference, I assume that the nearby table in the report should be produced like this:
![table-nearby](https://cloud.githubusercontent.com/assets/7351977/16863006/e7949c78-4a78-11e6-881a-5afa920705c4.PNG)
**note that** :
- the first sorting is by Fallout level
- the second sorting is by number of people affected
- above sorting also applied to airports affected

____
### Priotities for places and airports
Due to the limited space in the report (only about 7 rows available for nearby places result), there is another thing that I'd like to ask you. in regards of big event like the screenshot above,what is your preference in showing the affected places and airports? 
I mean, should we show 
- top 7 city result: 
![topcity](https://cloud.githubusercontent.com/assets/7351977/16863273/1b0ec9e6-4a7b-11e6-8ea5-1d225ca8c520.PNG)

- or we should show all of the airport first and then the city:
![airport first](https://cloud.githubusercontent.com/assets/7351977/16863276/2466e172-4a7b-11e6-99b1-f2f96bf03268.PNG)

- or only the nearest/highest level of airport only, and the rest are the city
![highest_airport_only](https://cloud.githubusercontent.com/assets/7351977/16863279/28e3faaa-4a7b-11e6-8475-627c2c57b93a.PNG)

___
### Another volcano in map extent
in the Merapi's case, there are actually several volcanoes that are shown in the map extent.
![volcanoes](https://cloud.githubusercontent.com/assets/7351977/16863424/94e2b6d2-4a7c-11e6-8d44-ce6c23fe6ecd.PNG)

do you prefer showing all of them or just filter them to only the one that caused the ash. or do you have any other preference regarding to this?

___
**CC** : @Charlotte-Morgan @timlinux @Gustry @lucernae @samnawi @fredychandra

**Comment by adelebearcrozier** [comment]
Hi @ivanbusthomi, thank you for all your excellent work on this. It is looks wonderful. My answers to your questions below. 

**Re: Sorting Area**

1) According to Table 2 in the reference below, the preferred order that will be shown in the report table are descending from high fallout level to the low one, right?  - CORRECT

2) So, using that reference, I assume that the nearby table in the report should be produced like this: - YES, PERFECT

3) note that: 
    -  the first sorting is by Fallout level - NOTED
    -  the second sorting is by number of people affected - NOTED
    -  above sorting also applied to airports affected - NOTED

**Re: Priorities for places and airports**

4) Option 2: "or we should show all of the airport first and then the city:" seems best because we experience airport closures with low levels of ash. 

**Re: Another volcano in map extent**
5) I think filtering to the volcano which is erupting would avoid confusion.

@heriwaseso - what are your thoughts on my answers above, do you agree???

**Comment by timlinux** [comment]
Hi @adelebearcrozier 

![inasafe4 0architecture](https://cloud.githubusercontent.com/assets/178003/16871645/47007dc4-4ab2-11e6-9b1f-0028311240e6.png)

This is an updated schematic of the moving parts. I am also adding a checklist below of our running tasks so that you can get an idea of current status:

- [x] implement a tool to extract a single time slice from NetCDF as GeoTIFF
- [x] Implement continuous ASH hazard on classified polygon landcover impact function
- [x] Implement continuous ASH hazard on continuous raster population impact function
- [ ] Implement continuous ASH hazard on places impact function
- [ ] Implement continuous ASH hazard on airports impact function
- [ ] collate impacts from the above four impact function into a single python data structure
- [x] create QGIS map composer template that matches the one provided by @adelebearcrozier 
- [x] render collated impact data into the template and generate a PDF
- [ ] install ubuntu fonts into realtime docker so that the template matches the one on @ivanbusthomi 's desktop
- [ ] implement the landing page for ASH realtime (with password protection)
- [ ] write system documentation describing the various components (as per diagram above)
- [ ] document map template tokens and tags so that @adelebearcrozier or other interested parties can potentially generate other map products
- [ ] document all the data used (exposure and context)
- [ ] write up deployment guide for running on a new server

Most of the above tasks are near completion, some (like docs) are in progress. We will update the above list as we check things off...

**Comment by lucernae** [comment]
[report.pdf](https://github.com/inasafe/inasafe/files/378008/report.pdf)
[report.pdf](https://github.com/inasafe/inasafe/files/378028/report.pdf)

I can finally put the overview map 💃 

Some things to think about:
- [ ] Make the ocean transparent in overview map - @ivanbusthomi 
- [ ] Right now maximum number of rows in nearby table is 6, with maximum number of airport 2. Sorted by hazard level, then population count, then place name. Sometimes airport name is really long (it contains text: International Airport) - CC @ivanbusthomi 
- [ ] Should we define max rows for landcover table too? Also sometimes it contains "Other" category, should we display it too? - CC @ivanbusthomi 
- [ ] Empty text for nearby and landcover table, in case no place affected (I think it is not likely for landcover since ash will always fall on the ground) - CC @ivanbusthomi

**Comment by ivanbusthomi** [comment]
@lucernae 
this is the data that you can use in the overview layer. it's a zip that contains shp files.
[overview.zip](https://github.com/inasafe/inasafe/files/378068/overview.zip)

I will update you here with other things that needed to be updated in the report.

**Comment by adelebearcrozier** [comment]
Hi @timlinux, @ivanbusthomi, @Charlotte-Morgan and others, this is really exciting progress. Many thanks. Please dont hesitate to let me know if you need anything from me or my colleagues at BG. Again, fantastic progress. Cheers.

**Comment by timlinux** [comment]
Thanks @adelebearcrozier 

If you would like to test the staging site, I think it would be useful to start getting direct feedback (ideally as updated subtasks to this ticket):

http://staging.realtime.inasafe.org/realtime/ash/

Please bear in mind that the above site is a development snapshot and not intended for production use yet. When we have sign off from you we will deploy to http://realtime.inasafe.org

Some additional tasks from my side:

- [ ] @adelebearcrozier could you supply us with some context text for the top paragraph in the page to replace 'lorem ipsum' ?
- [ ] @ivanbusthomi I think the place name labels are a little large, what about reducing them to 2/3 of the current size?
- [ ] @lucernae we still need to have access control restrictions for ash uploads, and then give @adelebearcrozier and anyone she nominates with the appropriate permissions to do uploads
- [ ] @ivanbusthomi and @adelebearcrozier we either need to get you set up with using the little tool that @ivanbusthomi wrote that will extract a single time slice form the NetCDF, or if possible you can integrate the logic I shared above into your product generation scripts to produce the single band tiff product for the ash thickness.

```
gdal_translate \
  -b 65 \
  -of Tiff \
  NETCDF:"/Users/timlinux/Downloads/soputan_forecast_2016.res.nc":THICKNESS thickness.tif
```

Expect to tweak the above as needed (e.g. its hard coded to band 65 above.

@adelebearcrozier any other feedback and inputs you can provide us with will be greatly appreciated!

**Comment by ivanbusthomi** [comment]
Hi @lucernae,
I have a few suggestions regarding the realtime report template. 
* For the event name, I think it would be good if we add "Volcano:" just before the volcano name. And also, I noticed that the Province section show Indonesia instead of province name
![1](https://cloud.githubusercontent.com/assets/7351977/17123898/51dea93e-5311-11e6-8f49-3ae7166f3215.PNG)
* In the potential impact table, how about changing the style of the values in People Affected and Ash thickness rows to be bold? To emphasise the value and make it look like the initial template as close as possible
![2](https://cloud.githubusercontent.com/assets/7351977/17123991/1614f682-5312-11e6-9ca5-0d5971508617.PNG)
* Nearby places and Land cover impact are looking great! what do you think about adding a thousand separators in the column? this is what I imagined it would look like
![3 1](https://cloud.githubusercontent.com/assets/7351977/17124089/cfcd2afe-5312-11e6-8765-ebc652adccc4.PNG)
* I suggest that we decrease the transparent value in the hazard layer so that the colour will look more like the one that we show in the potential impact table. Our initial value is 50% transparency, and my suggestion is around 15% transparency. 
![transparency](https://cloud.githubusercontent.com/assets/7351977/17124486/a370ccec-5315-11e6-90f8-7cbd5c0f43b8.png)

* I also noticed that not all first letter in each word of the airport text in the report is not capitalised. I think if we make it capital on the first letter of each word like this one below,the template will look a little bit better.
![4](https://cloud.githubusercontent.com/assets/7351977/17124333/8b4931c8-5314-11e6-8a2c-f51a68fa5af3.PNG)

* I also made a few tweaks to the '.qpt' file. Those tweaks are making the border of each box to be perfectly aligned to the right and left, making the map slightly bigger, and updating the disclaimer text to match the data sources. you can find the template and the HTML files in the "project_templates" BTsync folder.

This is my whole tweak and suggestions
![standard_example](https://cloud.githubusercontent.com/assets/7351977/17124384/ea61f014-5314-11e6-921c-ae3ed3394019.png)

**Comment by ivanbusthomi** [comment]
Hi @adelebearcrozier,
I have a few questions for you,

1. In the InaSAFE desktop analysis report, we once had an issue with the usage of medium or moderate. After a discussion, we then decided to use the term medium instead of moderate. According to @felix-yew , BNPB will likely to prefer using the term medium. What do you think about this? Do you prefer keeping the term "moderate" or is it okay to use the term "medium" in the realtime report and make it the same as the desktop analysis report?
 
2. I noticed that the ash thickness range that you provided us in the template is a little bit strange, because if I understand correctly there are some missing ranges. 
![5](https://cloud.githubusercontent.com/assets/7351977/17125046/639cebbe-531a-11e6-89bd-eee3404293b7.PNG)
Here is what I understand about the range.
  * 0 - 0.01             :point_left: missing range
  * __0.01 - 0.1__
  * 0.1 - 0.2            :point_left: missing range
  * __0.2 - 2__
  * 2 - 2.1               :point_left: missing range
  * __2.1 - 5__ 
  * 5 - 5.1               :point_left: missing range
  * __5.1 - 10__
  * __>10__

   Could you please give some explanation or background on this? I am not aware if the range is based on some analysis, but this is what I have in mind on how the range should look like:
  * 0 - 0.01 (Not affected - will not be shown in the report)
  * 0.01 - 0.1 (Very Low)
  * 0.1 - 2 (Low)
  * 2 - 5 (Medium)
  * 5 - 10 (High)
  * >10 (Very High) 

3. As @lucernae mentioned:
> Also sometimes it contains "Other" category, should we display it too?

   the data actually contains 6 classes that are the result of reclassifying the original land cover data that we got from BIG. The 6th class are the "other" class that actually contains values that we think will not suite into the other 5 class. Those values are:
  * Pasir / Bukit pasir laut (_Sand/Sand dunes_)
  * Semak belukar / Alang-alang (_Bush_)
  * Perairan lainnya (_other water area_)

   I personally prefer that the other class are not shown in the report since I don't think it has any economic values. But if we consider that it is still a landcover data, then we might as well show it in the report. What do you think on this matter?
---
__cc__ : @timlinux @Charlotte-Morgan @felix-yew @fredychandra @samnawi

**Comment by ivanbusthomi** [comment]
@timlinux 
>I think the place name labels are a little large, what about reducing them to 2/3 of the current size?

I will try updating the style in the BTsync folder. I will update the changes that I made here

>we either need to get you set up with using the little tool that @ivanbusthomi wrote that will extract a single time slice form the NetCDF, or if possible you can integrate the logic I shared above into your product generation scripts to produce the single band tiff product for the ash thickness.

FYI, I made some changes to my tool, I hope it would not be buggy again. 
it is here :point_right: https://github.com/ivanbusthomi/cdf_converter/releases/tag/v1.1.0
here is the update screenie
![v1 1 0](https://cloud.githubusercontent.com/assets/7351977/17126042/86b39086-5323-11e6-885a-2e5d26516b77.PNG)

**Comment by lucernae** [comment]
@ivanbusthomi 

Commenting out the issue from @ivanbusthomi :

> - [ ] For the event name, I think it would be good if we add "Volcano:" just before the volcano name. And also, I noticed that the Province section show Indonesia instead of province name 
I can add Volcano. However the province name is coming from the data. I look at it, and it doesn't contains province name. So, we should update the data first (or uses another?). I am using **GVP_Volcano_List_Darwin_VAAC_AOR_final.shp**.

> - [ ] Nearby places and Land cover impact are looking great! what do you think about adding a thousand separators in the column? this is what I imagined it would look like 
We will do that

> - [ ] I suggest that we decrease the transparent value in the hazard layer so that the colour will look more like the one that we show in the potential impact table. Our initial value is 50% transparency, and my suggestion is around 15% transparency.

We can change from the hazard.qml

> - [ ] I also noticed that not all first letter in each word of the airport text in the report is not capitalised. I think if we make it capital on the first letter of each word like this one below,the template will look a little bit better.

We will do that

> - [ ] I also made a few tweaks to the '.qpt' file. Those tweaks are making the border of each box to be perfectly aligned to the right and left, making the map slightly bigger, and updating the disclaimer text to match the data sources. you can find the template and the HTML files in the "project_templates" BTsync folder.

I probably should talk more in private channel about this.

I will start working on it, and I will checkmark the task after it was deployed.
I will give the credentials for staging.realtime.inasafe.org in the email.

**Comment by Charlotte-Morgan** [comment]
Hi Everyone and @adelebearcrozier 
Jumping in on the thresholds for analysis and the labels on the report.
Ash size ranges should be continuous and mutually exclusive; this means no gaps and no overlap.
eg
0 - 0.1
>0.1 - 2.0
> 2.0 - 5.0
>5.0 - 10
>10 

@adelebearcrozier - can you please clarify the classes you require as defaults in volcanic ash analysis?

**Comment by timlinux** [comment]
Hi @ivanbusthomi and @adelebearcrozier 

* @ivanbusthomi I really like the report variant with the blue backgrounds - lets see what @adelebearcrozier has to say about it?
* @adelebearcrozier Would it be possible to include the InaSAFE (black monochrome) logo somewhere? Perhaps in the box which describes the InaSAFE version used put pulled over to the right? Including our branding will be very good promotion for InaSAFE if the reports get passed around and used in other contexts outside of the InaSAFE realtime web site.
* @adelebearcrozier and @ivanbusthomi can I suggest to remove the frame borders from the last two sections (impact description and InaSAFE version)?

**Comment by adelebearcrozier** [comment]
@timlinux 

Text requested to replace Lorem Ipsum:

This page contains near realtime volcanic ash impact assessments generated prior to or during recent volcanic eruptions in the Indonesia region. Ashmaps of volcanic ash thickness are produced by Indonesia’s Geological Agency, Badan Geologi (http://www.bgl.esdm.go.id) and used by the Indonesian National Disaster Management Agency (BNPB) to produce InaSAFE impact assessments (http://www.bnpb.go.id or http://realtime.inasafe.org). This information is used by disaster managers to help them understand the potential scale of a volcanic disaster and to respond faster to the hardest hit communities.

**Comment by adelebearcrozier** [comment]
@ivanbusthomi, @timlinux, @Charlotte-Morgan, @lucernae  - Feedback as requested:

- Do you prefer keeping the term "moderate" or is it okay to use the term "medium" in the real-time report and make it the same as the desktop analysis report?
I see no problem using 'medium' instead of 'moderate' in order to keep consistency so please go ahead with this. 

- The volcanic ash thickness range missing values:
Apologies, not sure what happened there but definitely some typos. Your proposed breakdown is correct. Thank you and please use as below. 

    0 - 0.01 (Not affected - will not be shown in the report)
    0.01 - 0.1 (Very Low)
    0.1 - 2 (Low)
    2 - 5 (Medium)
    5 - 10 (High)
    >10 (Very High)

- the additional land cover classes 'other':
I agree, the land cover classes you mention are not essential and should be left out of the report. 

- New styling of the report (blue)
I think the styled version with the blue great. Lets go with it. 

- Adding InaSAFE monochrome logo
Absolutely, please fit it in where you can. 

- Removal of frame borders
Happy to remove 

Additional comments:
1. I will test the staging area today (including the tool for extracting the hazard layer) and get back to you.
2. I would like to draw your attention to the resource I have started using for the most up to date classification of volcano alert level in Indonesia. Was someone asking about how to assign this recently??? Anyway if you go to:  http://magma.vsi.esdm.go.id/ you will see a new web interface developed by BG which depicts the current volcano alert level for every volcano in Indonesia. You can select volcanoes to drill down into more information. For reference, their scheme is as follows:

GREEN: Normal (Alert Level I)
YELLOW: Waspada (Alert Level II)
ORANGE: Siaga (Alert Level III)
RED: Awas (Alert Level IV)

**Comment by timlinux** [comment]
hi @adelebearcrozier 

Thanks for your replies. Regarding your last comment of the magma web site:

Wow that looks really nice. For easy reference of those reading this thread later, here is what it offers:

![screen shot 2016-07-27 at 9 54 42 am](https://cloud.githubusercontent.com/assets/178003/17168042/7c70a484-53e0-11e6-9ed9-cd22cbdb5e2c.png)


* It would be really good to get the little fact sheet and webcam image linked into the ash realtime landing page table row (maybe a snapshot from the web cam at the time the report is generated, not the live feed). With @adelebearcrozier 's permission and the technology partner who made the site's permission of course. Then have them availble as little icons next to each ash entry on the landing page.

* @adelebearcrozier & @Charlotte-Morgan maybe we could also ask them to link to the InaSAFE report on ash realtime in their fact sheet? I guess that is something for a little later down the line.

@adelebearcrozier do you know if they publish a raw data feed of the volcano status data shown on the map?

**Comment by adelebearcrozier** [comment]
Hi @timlinux, it is fabulous isn't it, the site was developed and is maintained by Badan Geologi in house through the volcano monitoring division. I have been collaborating with them on the VONA functionality of the site (volcano observatory notice to aviation). They have an Android app version under development which will be launched publically through Google Play. They work mostly in XML and geojson formats. I am working with them to upload polygons of current volcanic ash advisories and satellite imagery of ash plumes so I think they would welcome links to the inasafe reports. I'd be happy to put you and @Charlotte-Morgan in touch with the team in Bandung.

**Comment by adelebearcrozier** [comment]
Hi @ivanbusthomi , regarding the following:

>FYI, I made some changes to my tool, I hope it would not be buggy again.
>it is here :point_right: https://github.com/ivanbusthomi/cdf_converter/releases/tag/v1.1.0
>here is the update screenie

Can you tell me if i need a particular version of QGIS for this to work?

**Comment by ivanbusthomi** [comment]
Hi @adelebearcrozier 
> Can you tell me if i need a particular version of QGIS for this to work?

QGIS 2.0 or higher should be just fine. Please let me know if you have any problem regarding the tools.

**Comment by ivanbusthomi** [comment]
Hi @timlinux and @adelebearcrozier 

This is my attempt adding the InaSAFE monochrome logo to the title, and removing the frame from the last two elements. What do you think?
![color_example_line](https://cloud.githubusercontent.com/assets/7351977/17356553/5ae9e8c8-5982-11e6-8244-1c060bd0d6e4.png)

___
__CC__ @Charlotte-Morgan @lucernae

**Comment by timlinux** [comment]
@ivanbusthomi I wanted to suggest putting it in the lower right of the report next to the line that says 'This report was created using InaSAFE 3.5....' rather. As you have done it above it looks a bit visually unbalanced to me. Could you try my suggestion and share a preview pic?

**Comment by ivanbusthomi** [comment]
Hi @timlinux 
I updated the template, and these are the result. The one on the left is based on your suggestion, and the one on the right is my attempt of putting the logo on top of the overview map.
![color_example_line_with_inasafe_logo-comparation](https://cloud.githubusercontent.com/assets/7351977/17471026/425a0dd2-5d6b-11e6-8295-fd81fa23a583.png)

@adelebearcrozier @lucernae what do you think about those two templates? I will update the template on the BtSync folder once it is approved.
___
__CC__ : @Charlotte-Morgan @samnawi @fredychandra

**Comment by timlinux** [comment]
@ivanbusthomi 👍 for the left hand version with the logo over the blue line at the bottom.

**Comment by Charlotte-Morgan** [comment]
@ivanbusthomi +1 for the left hand version with the logo over the blue line at the bottom

**Comment by Charlotte-Morgan** [comment]
@ivanbusthomi  - the winner is 1st Template - yay
please implement this - give it to @lucernae so he can upload all the experimental data to this template ready for feedback from BG colleagues and @adelebearcrozier when she is better

**Comment by ismailsunni** [comment]
@timlinux @adelebearcrozier 
I just knew yesterday that my friend in college is one of the developer in that magma website.

**Comment by adelebearcrozier** [comment]
@Charlotte-Morgan @timlinux @ivanbusthomi  - i also like 1st template :+1:

**Comment by adelebearcrozier** [comment]
@ismailsunni - is your college friend Anto??

**Comment by ismailsunni** [comment]
@adelebearcrozier yes, he is. We stayed in the same dormitory, so I know him quite well. :D

**Comment by adelebearcrozier** [comment]
@ismailsunni - thats great, I worked with Anto back in May in Bandung, what a lovely guy (and so good with all things html and android!)

**Comment by adelebearcrozier** [comment]
Hi @ivanbusthomi - i am using your cdf_converter in qgis to create a hazard file for generating a report. Had no problems installing or loading data, thanks for the clear instructions. 

However, my extracted tiff is black and has no values for thickness. 
Have i done something obviously wrong? 
I include screenshots of my inputs for the tool and the resulting tiff. 

![cdf](https://cloud.githubusercontent.com/assets/6599840/17578408/9a039e10-5fcb-11e6-8033-a1d4f0243ac9.png)
![raster](https://cloud.githubusercontent.com/assets/6599840/17578410/a0b3a066-5fcb-11e6-9811-702068ecf8d5.png)

**Comment by timlinux** [comment]
@adelebearcrozier I think it is a simple colour ramp scaling issue which you should be able to resolve by opening the raster properties, then choosing:

``Load min/max values`` then set to ``min/max`` then click ``load``

Here is a little animation to show it:

![globe](https://cloud.githubusercontent.com/assets/178003/18246059/5663ebb8-7369-11e6-9a02-03427aa4dbdc.gif)

**Comment by Charlotte-Morgan** [comment]
Hi everyone - thanks for all the hard work on this, contributions to the ticket and code.
I believe @adelebearcrozier is on leave for a while. Hopefully when she returns we can get an update on end user testing and implementation plans.

**Comment by Charlotte-Morgan** [comment]
Hi @lucernae - @adelebearcrozier is back and ready to test the realtime system but we are getting an error on the staging site. Is it possible for you to check that everything is happy and let us know?
Thanks :)

@timlinux

**Comment by adelebearcrozier** [comment]
Hi all, I'm back from leave and testing the real-time volcanic ash. I have successfully extracted a tiff file using Ivan's tool (thanks Ivan). I noticed the staging area is down and i will wait for confirmation that it is back then test. I am keen to then get BG testing ASAP.

**Comment by Charlotte-Morgan** [comment]
:)

**Comment by adelebearcrozier** [comment]
Hi @ivanbusthomi - thanks for the great CDF converter tool and the helpful animation on how to visualise in qgis. It works well for me. Just a side question, will the tool work if the data is in geographic rather than UTM? - just for future reference as sometimes we model in lat/lon when ash is expected to cross UTM boundaries. Again all works fine for UTM but if we have some output in lat/lon it would be good to be able to extract the tiff with a CRS like WGS84. I understand though if this creates flow on problems with the rest of the procedure and i might just flag it for a future update. Cheers

**Comment by adelebearcrozier** [comment]
@Charlotte-Morgan @timlinux @ivanbusthomi @fredychandra - tried to test again this morning but still unable to access the ash staging area.

**Comment by ivanbusthomi** [comment]
Hi @adelebearcrozier ,
glad to hear that the tool helps,
> Just a side question, will the tool work if the data is in geographic rather than UTM? - just for future reference as sometimes we model in lat/lon when ash is expected to cross UTM boundaries

The coordinate selector is to define the coordinate system of the output layer. From the NetCDF metadata, I only manage to get the extent of the data, not the coordinate system. So the coordinate system must be defined manually.
I think it should be working just fine with data in geographic system, but if you encountered some other problem feel free to let me know.

regarding the staging state, It has been offline for the last few days, and Mas @lucernae told me that he has been trying to fix the problem already. Let's hope that the problem will be fixed soon. 😄

**Comment by lucernae** [comment]
@adelebearcrozier I'm trying to use another server for staging. I'll get it back to you after I'm done.

**Comment by adelebearcrozier** [comment]
Thanks @ivanbusthomi for the information. And thanks @lucernae for your efforts.

**Comment by lucernae** [comment]
@adelebearcrozier @ivanbusthomi Just to let you know, you can use http://196.214.55.116:61100/realtime/ for testing. I spawned another staging server while we wait for our previous server fit again.

**Comment by Charlotte-Morgan** [comment]
hi @lucernae - that is great thank you. I was just talking to @adelebearcrozier and she is ready to test this for us right now - so perfect timing :)
Please tell me - when I click the link above I do not get the ash realtime test site but instead I get the eq platform - maybe not what I was expecting?

**Comment by lucernae** [comment]
Like in staging, you have to log in first. Click at Account > Login using this credentials (like the previous one in staging):

user: volcano@realtime.inasafe.org
pass: t3st4ccount

**Comment by adelebearcrozier** [comment]
Thanks @ivanbusthomi and @lucernae for all your efforts - i was able to create an event and see it appear on the map however no report was generated. I cant attach the TIFF here, but can you see the one i uploaded to let me know if there might be any issue with it?

**Comment by adelebearcrozier** [comment]
@ivanbusthomi , @lucernae - perhaps i am not understanding the procedure. I generated a hazard tiff > created a ash event and uploaded the tiff > it said a report was in the process of being generated > when i refreshed the page it told me "no report generated for this event". Hope that helps some more.

**Comment by ivanbusthomi** [comment]
Hi @adelebearcrozier ,
Sorry for the late reply. Just now I checked the temporary ash realtime that Mas @lucernae created, and I noticed that there are two new events already generated. Did you upload the Sinabung and Soputan data? 
![ash](https://cloud.githubusercontent.com/assets/7351977/19229221/2dae233e-8ef6-11e6-9a8a-06fbcbfb5ac4.PNG)

When I tried clicking the report button, it indeed says that no report generated. Right now I am trying replicating the steps that I did earlier, and confirming whether the report is in the wrong directory or not. I will give you an update soon.

Mas @lucernae , could you please give some information regarding this?

**Comment by lucernae** [comment]
I was inspecting the process and it turns out the process crashed midway.
I'll put the error here for diagnostic uses.

```
CallGDALError: <p>Error while executing the following shell command:</p><pre>"/usr/bin/gdalwarp" -q -t_srs EPSG:4326 -r near -tr 0.0016665999999999999 0.0016665999999999999 -cutline /home/realtime/inasafe/2016-10-10/root/work/extent_YQUpp2.kml -crop_to_cutline -ot Float64 -of GTiff "/home/realtime/analysis_data/ash/exposure/WorldPop_200m.tif" "/home/realtime/inasafe/2016-10-10/root/work/clip_RsQsg4.tif"</pre><p>Error message: Process crashed.
  File "realtime/tasks/ash.py", line 47, in process_ash
    hazard_url=hazard_url)
  File "realtime/ash/make_map.py", line 78, in process_event
    event.calculate_impact()
  File "realtime/ash/ash_event.py", line 629, in calculate
 _impact
    'population_impact')
  File "realtime/ash/ash_event.py", line 593, in calculate_specified_impact
    cell_size=cell_size)
  File "safe/utilities/clipper.py", line 124, in clip_layer
    raise e
```

**Comment by adelebearcrozier** [comment]
Hi @ivanbusthomi, yes my two were soputan and sinabung. Please disregard sinabung as it was meant to be soputan.

**Comment by ivanbusthomi** [comment]
@adelebearcrozier I tried recreating the steps that I took earlier, and I encountered the same message like yours. Let's wait for future update from Mas @lucernae 

Mas @lucernae thanks for the update, let's hope it's just a small bug :+1:

**Comment by adelebearcrozier** [comment]
Will do @ivanbusthomi - 👍👍

**Comment by lucernae** [comment]
I saw it works for Sangeang Api. Could you try it with various example?

**Comment by lucernae** [comment]
@adelebearcrozier and @ivanbusthomi It seems the hazard uploaded is not in WGS84 CRS. Note that the hazard needs to be in WGS 84 (EPSG:4326).

**Comment by adelebearcrozier** [comment]
@lucernae and @ivanbusthomi - that's interesting. I thought I had generated WGS84 UTM 51N using the CDF converter. Can you tell what CRS it is?  I will try to extract a new one tomorrow. Thanks for clarifying for me.

**Comment by lucernae** [comment]
@adelebearcrozier for Soputan, it is indeed WGS84 UTM 51N, but for the other sample I got like Sangeang Api, Merapi, etc, they were using EPSG 4326. So at first I thought the default would be EPSG 4326?
I tried converting the CRS using QGIS and this is the results:

http://196.214.55.116:61100/media/reports/ash/pdf/report_VE9GjIQ.pdf

**Comment by adelebearcrozier** [comment]
@lucernae, @ivanbusthomi - that report looks perfect. I will try to reproduce it tomorrow. Many thanks for finding the issue. I knew it would be my mistake:) cheers Adele

**Comment by lucernae** [comment]
That's ok. The point is, we should agree for the default CRS, so in the case the hazard file provided in different CRS, we can try to convert it to the default CRS.
There are still some bugs in the report I haven't been able to debug yet. For example, the mountain symbol was not rendered in the staging site. But somehow, it was rendered correctly in my local environment.

**Comment by ivanbusthomi** [comment]
@lucernae and @adelebearcrozier 
>I saw it works for Sangeang Api. Could you try it with various example?

The Sangeang Api actually in WGS 84. so it might be indeed what causing the report to not be created

**Comment by timlinux** [comment]
Folks this ticket has become very long and unwieldy - I am going to close it. Lets pick up further requirements in small specific tickets please. Thanks.
